build_tools/github_actions/summarize_test_pytorch_workflow.py

```diff
@@ -0,0 +1,123 @@
+    return platform.system() == "Windows"
+
+
+LINE_CONTINUATION_CHAR = "^" if is_windows() else "\\"
+LINE_CONTINUATION = f" {LINE_CONTINUATION_CHAR}\n  "
+
+
+def run(args: argparse.Namespace):
+    summary = ""
+
+    summary += "## PyTorch test environment reproduction instructions\n\n"
+
+    if not is_windows():
+        summary += "(Optional) Run under Docker\n"
+        summary += "```bash\n"
+        summary += "sudo docker run -it" + LINE_CONTINUATION
+        summary += "--device=/dev/kfd --device=/dev/dri" + LINE_CONTINUATION
+        summary += (
+            "--ipc=host --group-add=video --group-add=render --group-add=110"
+            + LINE_CONTINUATION
+        )
+        summary += "ghcr.io/rocm/no_rocm_image_ubuntu24_04:latest\n"
+        summary += "```\n\n"
+        summary += "```bash\n"
+        summary += "# Install extra packages\n"
+        summary += "sudo apt install python3.12-venv -y\n"
+        summary += "```\n\n"
+
+    pytorch_remote_name = "upstream" if args.pytorch_git_ref == "nightly" else "rocm"
+    pytorch_repo_org = "pytorch" if args.pytorch_git_ref == "nightly" else "ROCm"
+    summary += "Fetch pytorch source files, including tests\n\n"
+    summary += "* (A) Clone pytorch if starting fresh\n\n"
+    summary += "    ```bash\n"
+    summary += f"    git clone --branch {args.pytorch_git_ref} --origin {pytorch_remote_name} https://github.com/{pytorch_repo_org}/pytorch.git\n"
+    summary += "    ```\n\n"
+    summary += "* (B) Switch to pytorch ref using an existing repository\n\n"
+    summary += "    ```bash\n"
+    summary += "    cd pytorch\n"
+    summary += f"    git remote add {pytorch_remote_name} https://github.com/{pytorch_repo_org}/pytorch.git\n"
+    summary += f"    git fetch {pytorch_remote_name} {args.pytorch_git_ref} && "
+    summary += f"git checkout {pytorch_remote_name}/{args.pytorch_git_ref}\n"
+    summary += "    ```\n\n"
+
+    summary += "Install torch and test requirements into a venv\n\n"
+    summary += "```bash\n"
+    if is_windows():
+        summary += "python -m venv .venv && .venv\Scripts\Activate.bat\n"
+    else:
+        summary += "python3 -m venv .venv && source .venv/bin/activate\n"
+    summary += "pip install" + LINE_CONTINUATION
+    summary += f"--index-url={args.index_url}/{args.index_subdir}" + LINE_CONTINUATION
+    summary += "torch"
+    summary += f"=={args.torch_version}" if args.torch_version else ""
+    summary += "\n"
+    summary += "pip install -r pytorch/.ci/docker/requirements-ci.txt\n"
+    summary += "```\n\n"
+
+    summary += "## PyTorch testing instructions\n\n"
+    summary += "See [Running/testing PyTorch](https://github.com/ROCm/TheRock/tree/main/external-builds/pytorch#runningtesting-pytorch). "
+    summary += "For example:\n\n"
+    summary += "```bash\n"
+    summary += "PYTORCH_TEST_WITH_ROCM=1 python pytorch/test/run_test.py --include test_torch\n"
+    summary += "```\n"
+
+    gha_append_step_summary(summary)
```

Comment:
> 3\. related to the different markdown versions:
>         do workflow steps show proper markdown? i always thought it is just plain text.

This writes to `GITHUB_STEP_SUMMARY` (https://docs.github.com/en/actions/reference/workflows-and-actions/workflow-commands#adding-a-job-summary), which supports GitHub flavored markdown.

It will appear like here: https://github.com/ROCm/TheRock/actions/runs/19553166331#summary-55989478372 . Note that when there are multiple jobs in the same workflow run, we will have a list like here: https://github.com/ROCm/TheRock/actions/runs/19680190301#summary-56371979719.